autoload/magit/utils.vim

```diff
@@ -20,12 +20,17 @@ function! magit#utils#is_submodule(dirname)
 	return ( index(s:submodule_list, a:dirname) != -1 )
 endfunction
 
-" s:magit_cd_cmd: plugin variable to choose lcd/cd command, 'lcd' if exists,
-" 'cd' otherwise
-let s:magit_cd_cmd = exists('*haslocaldir') && haslocaldir() ? 'lcd ' : 'cd '
-" magit#utils#lcd: helper function to lcd. use cd if lcd doesn't exists
-function! magit#utils#lcd(dir)
-	execute s:magit_cd_cmd . fnameescape(a:dir)
+" magit#utils#chdir will change the directory respecting
+" local/tab-local/global directory settings.
+function! magit#utils#chdir(dir)
+  " This is a dirty hack to fix tcd breakages on neovim.
+  " Future work should be based on nvim API and has a `if has('nvim')` check.
+  if has('nvim')
+    let chdir = haslocaldir() ? 'lcd' : haslocaldir(-1, 0) ? 'tcd' : 'cd'
```

Comment:
`:tcd` has been introduced right after nvim 0.1.3 (I did not check, it should be officially present since nvim 0.1.4).

Before `:tcd` introduction, `haslocaldir(-1, 0)` will return an error:
```
E118: Too many arguments for function: haslocaldir                                                                                                                                                                                                                                
E15: Invalid expression: haslocaldir(-1, 0)                                                                                                                                                                                                                                       
```